src/pages/AddItem.js

```diff
@@ -1,3 +1,61 @@
+import { useState } from 'react';
+import { db } from '../lib/firebase';
+// import { useCollection } from 'react-firebase-hooks/firestore';
+
 export default function AddItem(props) {
-  return <h1>This is the form</h1>;
+  const [itemName, setItemName] = useState('');
+  const [purchaseFrequency, setPurchaseFrequency] = useState(0);
+  const [lastPurchased] = useState(null);
+
+  const handleName = (e) => {
+    setItemName(e.target.value);
+  };
+
+  const handleFrequency = (e) => {
+    setPurchaseFrequency(e.target.value);
+  };
+
+  const createListItem = (e) => {
+    e.preventDefault();
+    db.collection('generated_token')
+      .add({
+        item_name: itemName,
+        purchase_frequency: parseInt(purchaseFrequency),
+        last_purchased: lastPurchased,
+      })
+      .then((documentReference) => {
+        console.log('document reference ID', documentReference.id);
+      })
+      .catch((error) => {
+        console.log(error.message);
+      });
+  };
+
+  return (
+    <>
+      <h1>Add Item</h1>
+      <form onSubmit={(e) => createListItem(e)}>
```

Comment:
Events are passed by default so you can do the following.
```suggestion
      <form onSubmit={createListItem}>
```
Only when you need to explicitly pass a value then you would use an arrow function.
```
<form onSubmit={(value) => createListItem(value)}>
```

Reply:
Thanks Yenly! You explained this really well. I've always wondered why you see it one way vs. the other. Making those changes now!